### PR TITLE
Add BatchNorm2D layer

### PR DESCRIPTION
**Pull request by jessegrabowski**
WIP, still needs to track the running mean/variance of the data, plus a rewrite to swap out the batch computation for the running stats during prediction

### CODE REVIEW

File: pytensor_ml/pytensorf.py [hunk 1]
```diff
@@ -5,10 +5,35 @@
...
 from pytensor import Mode
 from pytensor.compile import Function, SharedVariable, get_mode
+from pytensor.compile.builders import OpFromGraph
 from pytensor.graph import FunctionGraph, RewriteDatabaseQuery, rewrite_graph
+from pytensor.graph.basic import equal_computations
 from pytensor.tensor.variable import Variable
 
-from pytensor_ml.rewriting.layers import predict_db
+
+class HashableOpFromGraph(OpFromGraph):
```
Comment (ricardoV94) [inline pytensor_ml/pytensorf.py:14]: Lets use a subclass of OpFromGraph `LayerOp`? And make sure all subclasses have props that can be used for op equality/merging.

I imagine number and ndim of nin/nout covers most of ir. Eg if two linear ops have the same number of inputs terms and dimensionality they will perform the same computation and the inner graph can be reused.

**Review by ricardoV94 — CHANGES_REQUESTED** [review]

File: pytensor_ml/pytensorf.py (hunk 1 above)
Comment (jessegrabowski) [reply pytensor_ml/pytensorf.py:14]: Right now we're doing `equal_computation`, you're saying just check the props instead?

Comment (ricardoV94) [reply pytensor_ml/pytensorf.py:14]: If you add props you don't need anything, PyTensor will generate the hash and equality methods automatically from those

**Comment by jessegrabowski** [comment]
This PR ended up touching a lot more than just batch norm, so I reorganized the commits to be logical chunks, and I won't squash it.
